lib/hal_api/errors.rb

```diff
@@ -4,10 +4,18 @@ module HalApi::Errors
 
   class ApiError < StandardError
     attr_accessor :status
+    attr_accessor :hint
 
-    def initialize(message = nil, status = 500)
+    def initialize(message = nil, status = nil, hint = nil)
       super(message || "API Error")
-      self.status = status
+      self.status = status || 500
+      self.hint = hint
```

Comment:
It's really up to the rails app how to handle this ... but I'm finding it useful to pass along a human readable "hint" in the error json.  To help track down problems more quickly.